receiver_test.go

```diff
@@ -0,0 +1,94 @@
+package gitkit_test
+
+import (
+	"testing"
+
+	"github.com/sosedoff/gitkit"
+	"github.com/stretchr/testify/assert"
+)
+
+type gitReceiveMock struct {
+	name            string
+	masterOnly      bool
+	allowedBranches []string
+	ref             string
+	isErr           bool
+}
+
+func TestMasterOnly(t *testing.T) {
+	testCases := []gitReceiveMock{
+		{
+			name:       "push to master, no error",
+			masterOnly: true,
+			ref:        "refs/heads/master",
+			isErr:      false,
+		},
+		{
+			name:       "push to a branch, should trigger error",
+			masterOnly: true,
+			ref:        "refs/heads/branch",
+			isErr:      true,
+		},
+	}
+
+	for _, tc := range testCases {
+		r := &gitkit.Receiver{
```

Comment:
Should use `t.Run(name)` here since your test cases include names. Also, make sure to include the actual `error` instead of `isErr` in the test cases so you can do `assert.Equal` without having to check if error is returned first.

Reply:
Makes sense, I think my latest push addresses this.